core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1178,6 +1178,37 @@ abstract class RDD[T: ClassTag](
     }
   }
 
+  /**
+   * Computing desired tree aggregate depth necessary to avoid exceeding
+   * driver.MaxResultSize during aggregation.
+   * Based on the formulae: (numPartitions)^(1/depth) * objectSize <= DriverMaxResultSize
+   * @param aggregatedObjectSizeInMb the size, in megabytes, of the object being tree aggregated
+   * @param numPartitions the number of partitions for which to aggregate partial results
+   * @param maxDriverResultSizeInMb the value of the parameter spark.driver.maxResultSize in MB
+   */
+  def getTreeAggregateIdealDepth(
+      aggregatedObjectSizeInMb: Int,
+      numPartitions: Int = getNumPartitions,
+      maxDriverResultSizeInMb: Int = Utils.memoryStringToMb(
+        context.getConf.get("spark.driver.maxResultSize"))): Int = {
```

Comment:
This value needs to be the result of `conf.get(MAX_RESULT_SIZE)`

Reply:
Indeed, more concise way of getting that. Thanks for pointing me to the config package object, didn't know about this one !

Reply:
So,playing in UTests with different ways to play with this config field made me realize something...
I tried the following:
*conf.set(""spark.driver.maxResultSize", "10g")* Then I get:
- *conf.get[Long](MAX_RESULT_SIZE)* returns 10737418240 -> it's in Bytes
- *Utils.memoryStringToMb(conf.get("spark.driver.maxResultSize"))* returns 10240 -> It's in megabytes 

So the units are as expected, but the results are a bit different from each other...

-> I'm not sure what's the best way to get the maxResultSize in Mb from code here:
- Utils.memoryStringToMb(conf.get("spark.driver.maxResultSize"))
OR
- conf.get(MAX_RESULT_SIZE) / (1000L * 1000L * 1000L)

??
(Knowing that these might yield slightly different results, but I don't know if it's a big deal)

Reply:
This may be moot as I don't think you need to check this value, but what's the issue - memoryStringToMb does return megabytes. Actually, we have an unfortunate conflation of megabyte and mebibyte here, but it's not material for purposes here.

Reply:
I figured that there's no issue at all there. In fact, 10737418240 / (1024 * 1024) == 10240